packages/pigeon/example/app/lib/src/event_channel_messages.g.dart

```diff
@@ -11,6 +11,65 @@ import 'dart:typed_data' show Float64List, Int32List, Int64List, Uint8List;
 import 'package:flutter/foundation.dart' show ReadBuffer, WriteBuffer;
 import 'package:flutter/services.dart';
 
+bool _listEquals(List<Object?>? list1, List<Object?>? list2) {
+  if (list1 == list2) {
+    return true;
+  }
+  if (list1 == null || list2 == null) {
+    return false;
+  }
+  if (list1.length != list2.length) {
+    return false;
+  }
+  bool elementsMatch = true;
+  for (int i = 0; i < list1.length; i++) {
+    if (list1[i] is List) {
+      elementsMatch =
+          _listEquals(list1[i] as List<Object?>?, list2[i] as List<Object?>?);
+    } else if (list1[i] is Map) {
+      elementsMatch = _mapEquals(list1[i] as Map<Object?, Object?>?,
+          list2[i] as Map<Object?, Object?>?);
+    } else {
+      elementsMatch = list1[i] == list2[i];
+    }
```

Comment:
The logic here is the same as below, so you can extract a helper. And you can also simplify the helper with patterns:

```dart
bool _deepEquals(Object? a, Object? b) {
  return switch ((a, b)) {
    (List a, List b) => _listEquals(a, b),
    (Map a, Map b) => _mapEquals(a, b),
    _ => a == b,
  };
}
```

Then everything here after the initial checks becomes:
```dart
  for (int i = 0; i < a.length; i++) {
    if (!_deepEquals(a[i], b[i])) {
      return false;
    }
  }
  return true;
```